src/jdk.compiler/share/classes/com/sun/tools/javac/main/JavaCompiler.java

```diff
@@ -1647,6 +1639,16 @@ public void visitSwitchExpression(JCSwitchExpression tree) {
             if (shouldStop(CompileState.LOWER))
                 return;
 
+            if (scanner.hasLambdas) {
```

Comment:
This is another bit I'm not sure about - why do we need a scanner to check if there are lambdas? I think we added this because there were concerns over running "too many visitors", but today we have many more, and we don't have scanners for them, so... (again, this is something for another PR)